cmd/erasure-server-pool.go

```diff
@@ -294,11 +295,30 @@ func (z *erasureServerPools) GetRawData(ctx context.Context, volume, file string
 }
 
 // Return the disks belonging to the poolIdx, and setIdx.
-func (z *erasureServerPools) GetDisks(poolIdx, setIdx int) ([]StorageAPI, error) {
-	if poolIdx < len(z.serverPools) && setIdx < len(z.serverPools[poolIdx].sets) {
-		return z.serverPools[poolIdx].sets[setIdx].getDisks(), nil
+func (z *erasureServerPools) GetDisks(poolIdx, setIdx int, local bool) (disks []StorageAPI, err error) {
+	if poolIdx >= 0 {
+		if poolIdx > len(z.serverPools) || setIdx > len(z.serverPools[poolIdx].sets) {
+			return nil, fmt.Errorf("Matching pool %s, set %s not found", humanize.Ordinal(poolIdx+1), humanize.Ordinal(setIdx+1))
+		}
+		disks = z.serverPools[poolIdx].sets[setIdx].getDisks()
+	} else {
+		// poolIdx < 0 means collect all disks in this cluster
+		for pool := range z.serverPools {
+			for set := range z.serverPools[pool].sets {
+				disks = append(disks, z.serverPools[pool].sets[set].getDisks()...)
+			}
+		}
 	}
```

Comment:
just use globalLocalDrives why do you need this?

Reply:
I like calling object-api.GetDisks() because we know for sure that disks are ready but I see know globalLocalDrives are initialized very early in the code, I will use it